src/core/controllers/viewport-controls.js

```diff
@@ -333,50 +361,66 @@ export default class ViewportControls {
     const isZoomOut = this.isFunctionKeyPressed(event);
 
     const newViewportState = this.viewportState.zoom({pos, scale: isZoomOut ? 0.5 : 2});
-    return this.updateViewport(newViewportState);
+    return this.updateViewport(newViewportState, LINEAR_TRANSITION_PROPS);
   }
 
+  /* eslint-disable complexity */
+  // Default handler for the `keydown` event
   _onKeyDown(event) {
-    if (this.viewportState.isDragging) {
-      return;
+    if (!this.keyboard) {
+      return false;
     }
-
-    const KEY_BINDINGS = {
-      w: 'moveForward',
-      W: 'moveForward',
-      ArrowUp: 'moveForward',
-
-      s: 'moveBackward',
-      S: 'moveBackward',
-      ArrowDown: 'moveBackward',
-
-      a: 'moveLeft',
-      A: 'moveLeft',
-      ArrowLeft: 'moveLeft',
-
-      d: 'moveRight',
-      D: 'moveRight',
-      ArrowRight: 'moveRight',
-
-      '=': 'zoomIn',
-      '+': 'zoomIn',
-
-      '-': 'zoomOut',
-
-      '[': 'moveDown',
-      ']': 'moveUp'
-    };
-
-    // keyCode is deprecated from web standards
-    // code is not supported by IE/Edge
-    const key = event.key;
-    const handler = KEY_BINDINGS[key];
-    if (this.viewportState[handler]) {
-      const newViewportState = this.viewportState[handler]();
-      this.updateViewport(newViewportState);
+    const funcKey = this.isFunctionKeyPressed(event);
+    const {viewportStateProps} = this;
+    let newViewportState;
+
+    switch (event.srcEvent.keyCode) {
```

Comment:
The old implementation of this key handling function is more clearer, why do we change it?